### ISSUE TITLE
minikube start option "--host-only-cidr" no longer works

### ISSUE THREAD

**Issue by jgoeres** [kubernetes/minikube#7063]
With my ancient minikube version (1.4.0), I had been using the option

    --host-only-cidr='192.168.99.1/24' 

which now gives me

    X Unable to start VM. Please investigate and run 'minikube delete' if possible: driver start: Error 
    setting up host only network on machine start: invalid CIDR address: '192.168.99.1/24'

with both 1.8.1 and 1.8.2.
I used the option to make sure that the minikube VM gets attached to the right host-only network. It now seemed to work fine without it, so this might be no longer needed, but still...

Regards
J

**Comment by tstromberg** [comment]
Can you please share the entire `minikube start` output and command-line flags used?

The example `--host-only-cidr` command works for me on v1.8.3 using hyperkit and virtualbox drivers on macOS.

**Comment by afbjorklund** [comment]
It seems like the quotes ended up _inside_ the actual configuration, somehow.

```go
package main

import "net"
import "fmt"


func main() {

	hostOnlyCIDR := "'192.168.99.1/24'"
	ip, network, err := net.ParseCIDR(hostOnlyCIDR)
	if err != nil {
		panic(fmt.Sprintf("%v", err))
	}

	fmt.Printf("ip: %v, network: %v\n", ip, network)
}
```

panic: invalid CIDR address: '192.168.99.1/24'

With just the regular quotes (i.e. not the embedded single quotes), it is fine.

ip: 192.168.99.1, network: 192.168.99.0/24

**Comment by afbjorklund** [comment]
Unless I am missing something, that is the default option anyway ?

`      --host-only-cidr='192.168.99.1/24': The CIDR to be used for the minikube VM (virtualbox driver only)`

**Comment by jgoeres** [comment]
I usually used to run

    minikube start --disk-size 30G --cpus 6 --insecure-registry <ourInternalRegistry> --memory 12g --host-only-cidr='192.168.99.1/24'

I don't remember the details why I actually started using the host-only-cidr switch - IIRC back when I started using minikube (around v1.4.0) the minikube VM sometimes ended up using the wrong host-only-network.
In the current version I am fine without the switch, cause, as you said, the default works just fine.

**Comment by priyawadhwa** [comment]
As @afbjorklund commented, I think the only thing to do here is to trim the address of quotes & validate that it is valid. If invalid, return an error message.

```
        hostOnlyCIDR = strings.Trim(hostOnlyCIDR, "'\"")
	ip, network, err := net.ParseCIDR(hostOnlyCIDR)
	if err != nil {
		return errors.Wrap(err, "invalid address")
	}
```

this code would probably live in a helper function that would be called here:

https://github.com/kubernetes/minikube/blob/39958f24e61e991630ca5a01eb1421110bec3caf/pkg/minikube/registry/drvs/virtualbox/virtualbox.go#L53


If anyone is interested in taking this on, feel free to take this issue by commenting `/assign`

**Comment by medyagh** [comment]
@Aut0R3V are you still working on this ?

**Comment by vigothehacker** [comment]
Hi, I was trying to replicate this issue. The value of fmt.Println(viper.GetString("host-only-cidr"))  that I am getting does not have the single quotes on my output hen running the command `./out/minikube start --driver=virtualbox -p virt --host-only-cidr='192.168.99.1/24'` and the minikube clusters starts as expected

**Comment by tharun208** [comment]
yes with the latest minikube I am also able to get the cluster started as expected

**Comment by nirs** [comment]
> With my ancient minikube version (1.4.0), I had been using the option
> 
> ```
> --host-only-cidr='192.168.99.1/24' 
> ```

The quotes are not needed since the value does not contain any special characters.

On Linux/macOS the unneeded quotes are stripped by the shell and the program never see them.

All these are the same:

- `--host-only-cidr=192.168.99.1/24`
- `--host-only-cidr='192.168.99.1/24'`
- `--host-only-cidr="192.168.99.1/24"`
- `'--host-only-cidr=192.168.99.1/24'`
- `"--host-only-cidr=192.168.99.1/24"`

The program will see the string: `--host-only-cidr=192.168.99.1/24`.

On Windows cmd.exe single quotes are literal and are passed to the program.

This is a user error - using Linux/macOS quoting on windows cmd.exe does not work - user should fix the way they run minikube.
